packages/typescript/src/index.ts

```diff
@@ -110,7 +116,8 @@ export default function typescript(options: RollupTypescriptOptions = {}): Plugi
       const resolved = resolveModule(importee, containingFile);
 
       if (resolved) {
-        if (resolved.extension === '.d.ts') return null;
+        if (resolved.extension.startsWith('.d.') && resolved.extension.match(/\./g)?.length === 2)
```

Comment:
I think we should allow files with multiple `.`s, since `file.with.lots.of.periods.ts` is a valid path

Reply:
fixed/improved.